airbyte-integrations/connectors/source-netsuite-odbc/source_netsuite_odbc/scripts/discover_cache.py

```diff
@@ -0,0 +1,175 @@
+          table_incremental_column = self.get_column_name(column)
+        properties[self.get_column_name(column)] = self.get_column_type(column)
+
+      json_schema = {
+        "$schema": "http://json-schema.org/draft-07/schema#",
+        "type": "object",
+        "properties": properties,
+      }
+
+      is_incremental_table = table_incremental_column is not None
+
+      result = {
+        "json_schema": json_schema,
+        "name": table_name,
+        "supported_sync_modes": ["full_refresh", "incremental"] if is_incremental_table else ["full_refresh"],
+        "source_defined_cursor": is_incremental_table,
+        "default_cursor_field": [table_incremental_column] if is_incremental_table else [],
+        "primary_key": primary_key_column,
+      }
+
+      return result
+    
+    def get_columns(self, table: json) -> Iterable[Mapping[str, Any]]:
+      table_name = self.get_table_name(table)
+      self.cursor.execute(f"SELECT table_name, column_name, type_name, oa_userdata FROM OA_COLUMNS WHERE TABLE_NAME = '{table_name}'")
+      # self.cursor.execute(f"SELECT COUNT(*) FROM OA_COLUMNS WHERE oa_userdata LIKE 'C%'")
+
+      while True:
+        row = self.cursor.fetchone()
+        # print(row)
+        if not row:
+          break
+        yield row
+
+    def is_column_custom(self, column):
+      oa_userdata = column[3]
+      if oa_userdata is None:
+        return False
+      return oa_userdata.startswith('C')
+
+    def get_column_name(self, column: json) -> str:
+      return column[1]
+    
+    def get_column_type(self, column: json) -> str:
+      netsuite_data_type = column[2]
+      return self.data_type_switcher[netsuite_data_type]
+    
+    def is_column_incremental(self, column: json) -> bool:
+      column_meta_data = column[3]
+      column_name = self.get_column_name(column)
+      if (column_meta_data is not None and 'M-' in column_meta_data) or column_name in ['datelastmodified', 'lastmodified', 'lastmodifieddate']:
+        return True
+      else:
+        return False
+      
+    def find_primary_key_for_table(self, table: json) -> str or None:
+      table_name = self.get_table_name(table)
+      self.cursor.execute(f"SELECT * FROM OA_FKEYS WHERE PKTABLE_NAME = '{table_name}'")
+      row = self.cursor.fetchone()
+      if row is None:
+        return None
+      return row[3] # return the column name for primary key
+
+
+def read_config_from_secrets():
+  parent_directory = os.path.dirname(os.path.dirname(os.path.realpath(__file__)))
+  parent_directory_of_parent = os.path.dirname(parent_directory)
+  with open(parent_directory_of_parent + "/secrets/config.json") as fp:
+    file_contents = fp.read()
+    return json.loads(file_contents)
+
+def print_to_file(schema):
+  parent_directory = os.path.dirname(os.path.dirname(os.path.realpath(__file__)))
+  with open(parent_directory + f"/schemas/{schema['name']}.json", 'w') as fp:
+    json.dump(schema, fp)
+  print('done writing file')
+
+
+def main():
```

Comment:
what is the script used for? when would we need to use it?

Reply:
Removing.  This is from when we thought we'd  need to hard code standard tables